cpp/src/arrow/util/tracing_internal.cc

```diff
@@ -184,6 +188,41 @@ opentelemetry::trace::Tracer* GetTracer() {
   return tracer.get();
 }
 
+int parseLine(char* line){
+  // This assumes that a digit will be found and the line ends in " Kb".
+  int i = strlen(line);
+  const char* p = line;
+  while (*p <'0' || *p > '9') p++;
+  line[i-3] = '\0';
+  i = atoi(p);
+  return i;
+}
+
+size_t GetMemoryUsedByProcess() { //Note: this value is in KB!
+  FILE* file = fopen("/proc/self/status", "r");
+  size_t result = -1;
+  char line[128];
+
+  while (fgets(line, 128, file) != NULL){
+    if (strncmp(line, "VmRSS:", 6) == 0){
+      result = parseLine(line);
+      break;
+    }
+  }
+  fclose(file);
+  return result*1000;
+}
+
+size_t GetMemoryUsed() {
+  size_t total_memory_size;
+  size_t used_memory_size;
+  struct sysinfo si;
+  sysinfo(&si);
```

Comment:
There was a portable implementation as part of a PR here: https://github.com/apache/arrow/pull/11426/files#diff-5d7d9a549780da2ec4baba08a43db3e1d524b70d6e8ae4f29cc8bfe831357f9cR178-R199 

We didn't end up pursuing the PR but you might be able to borrow some from that particular function.

That being said, I wonder if it would be more valuable to report `default_memory_pool()->bytes_allocated()`.  `sysinfo` is going to give us statistics on the entire server and `freeram` is sometimes misleading (e.g. if there are a lot of dirty pages then `freeram` could be high while at the same time we might be thrashing swap).

Reply:
@westonpace I get the bytes from the memory pool in the first span, but in the next ones I extracted the statistics from the sysinfo for report the RSS, I looked the PR which you mentioned, and there the method return just the FreeRam and for calculate the Current Ram in used I still need to use sysinfo.total_memory, but after that PR get merged I can add another function for get that, and don't use system includes here.

Reply:
Ah, yes, I see now that you are grabbing all three numbers.  `GetMemoryUsed` is less interesting than `GetMemoryUsedByProcess` in my opinion but I don't think it would hurt to report it.

> but after that PR get merged I can add another function for get that, and don't use system includes here.

That PR that I linked to will not be merged.  We ended up pursuing a different approach.  So we don't want to wait on it.  However, you're welcome to copy the approach that was being taken for this method.